Modules/Sources/AsyncImageKit/Helpers/FaviconService.swift

```diff
@@ -55,6 +55,20 @@ public actor FaviconService {
         task.task.cancel()
         tasks[key] = nil
     }
+
+    /// Parses HTML data to extract the apple-touch-icon URL or falls back to /favicon.icon
+    public static func makeFavicon(from data: Data, siteURL: URL) -> URL {
+        let html = String(data: data, encoding: .utf8) ?? ""
+        let range = NSRange(location: 0, length: html.utf16.count)
+        if let match = regex?.firstMatch(in: html, options: [], range: range),
+           let matchRange = Range(match.range(at: 1), in: html),
+            let faviconURL = URL(string: String(html[matchRange]), relativeTo: siteURL) {
```

Comment:
I don't spot any real issue here, but I think the casting between Range and NSRange, String and NSString, and potentially unsafe subscript, could be avoided by using `Swift.Regex`.